### PR TITLE
Add Blackboard

### PR DESCRIPTION
**Pull request by askmrsinh**
Blackboard OAuth Overview: https://docs.blackboard.com/rest-apis/learn/getting-started/3lo
Blackboard Developer API reference: https://developer.blackboard.com/portal/displayApi

### CODE REVIEW

**Comment by askmrsinh** [comment]
@atymic @lucasmichot

File: src/Blackboard/Provider.php [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    {
+        return ['subdomain'];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getAccessTokenResponse($code)
+    {
+        $basic_auth = base64_encode($this->clientId.':'.$this->clientSecret);
```
Comment (atymic) [inline src/Blackboard/Provider.php:40]: Can we use the guzzle basic auth param here?
https://docs.guzzlephp.org/en/stable/request-options.html#auth

File: src/Blackboard/Provider.php [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+        return json_decode($response->getBody(), true);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getTokenUrl()
+    {
+        return "https://{$this->getConfig('subdomain')}.blackboard.com/learn/api/public/v1/oauth2/token";
```
Comment (atymic) [inline src/Blackboard/Provider.php:54]: sprintf here please

Comment (atymic) [reply src/Blackboard/Provider.php:54]: (and in other spots in the PR)

**Review by atymic — CHANGES_REQUESTED** [review]

**Comment by atymic** [comment]
Also missing entry in the split script (monorepobuilder.yml)

File: src/Blackboard/Provider.php [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+     */
+    public function getAccessTokenResponse($code)
+    {
+        $basic_auth = base64_encode($this->clientId.':'.$this->clientSecret);
+        $response = $this->getHttpClient()->post($this->getTokenUrl(), [
+            RequestOptions::HEADERS     => ['Authorization' => 'Basic '.$basic_auth],
+            RequestOptions::FORM_PARAMS => $this->getTokenFields($code),
+        ]);
+
+        return json_decode($response->getBody(), true);
```
Comment (lucasmichot) [inline src/Blackboard/Provider.php:46]:
```suggestion
        return json_decode((string) $response->getBody(), true);
```

**Review by lucasmichot — APPROVED** [review]
Once @atymic comments have been taken in consideratiion, this will be GTM

**Review by lucasmichot — APPROVED** [review]
GTM @atymic

**Review by atymic — APPROVED** [review]

**Pull request merged**